### PR TITLE
Use datepicker in request edit form rather than simple string inputs

### PR DESCRIPTION
**Pull request by keathleydavidj**
## Purpose
Rather than typing in an arbitrary string as the date values for a vacation request, a user is going to expect the more familiar and intuitive interface of a datepicker for selecting dates.

## Approach

I've chosen to pull in the third party library "react-day-picker" to use their date picker components.
In this PR I've also taken the opportunity to generalize the editing form to avoid duplication across the create and edit routes. I also included the tests I forgot to import for my last PR, haha

## Learning
During the course of this section of work, there were several points where I found myself stuck trying to debug some weird obscure thing or another. When I pulled in outside help for these parts, one thing I noticed that was key to working through the issues was that no matter how simple the assumptions you're making may seem, it's important to go through them and verify that your perceptions of the situation match with reality. Also it can really help to simply leave the room and clear your head when you're stuck on something.

Changing from using regular strings to proper date object meant figuring out how to do post/pre processing for de/serialization to convert those dates for sending over the wire.

Something I was not expecting was that when comparing form states (for disabling buttons etc.) that using date objects meant that strict equality is broken with different instances of date (even with the same value).  This means that if the date gets changed from the initial value, but then changed back to the _same day_, the form was registering that as a changed state when you wouldn't expect that.

Another issue we tackled was how providing the form component with an empty request object while waiting for the real data to load from the api would simply not pay attention to the new data once loaded.  The way we got around this was by incrementing a counter anytime we load or update the data and passing this as the `key` to the component.  This would cause a new instance of the component to be rendered in place, with the initial data in place as expected.

Lastly, I ran into a bit of trouble with CI's timezone difference causing tests to fail.
As part of addressing this, we've decided to store the dates in the db in UTC timezone then adjust the time by the user's timezone offset when presenting in the UI.
Also set CI's timezone to CST to avoid more headache.

### CODE REVIEW

**Comment by taras** [comment]
@keathleydavidj please add to learnings, what you learned about working with serializable data, the pattern for passing that data to the component and checking equality of this data.

**Comment by taras** [comment]
Also, you can add to learnings how to use key to re-initialize the component

**Review by taras — APPROVED** [review]
👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ jobs:
     docker:
       # specify the version you desire here
       - image: circleci/node:8-browsers
+        environment:
+          TZ: "America/Chicago"
```
Comment (cherewaty) [inline .circleci/config.yml:12]: Nice, I didn't know this was possible!

**Review by cherewaty — APPROVED** [review]

**Pull request merged**